### PR TITLE
adding the ability to output the dataset

### PR DESCRIPTION
**Pull request by jkobject**
Hello Sergey, Alex,

I just realized I had made one more update to .mapped() in which I am adding the ability to output the dataset. I need it to remove batch effects in my model like scGPT does.

Let me know if that is ok. Once this is updated I can use this version :)

### CODE REVIEW

**Comment by falexwolf** [comment]
Thanks @jkobject! @Koncopd could you take a look at this so that it gets into the new release?

File: lamindb/core/_mapped_collection.py [hunk 1]
```diff
@@ -79,6 +79,7 @@ def __init__(
         unknown_label: Optional[Union[str, Dict[str, str]]] = None,
         cache_categories: bool = True,
         parallel: bool = False,
+        add_dataset_id: bool = False,
```
Comment (falexwolf) [inline lamindb/core/_mapped_collection.py:82]: @Koncopd - I don't fully understand what this does, but likely it deserves a better name.

**Comment by Koncopd** [comment]
Hm, maybe return this by default without any additional arguments? As we return a dict now, it is not a problem if we add some additinal keys.

**Comment by falexwolf** [comment]
Sounds like the right thing, Sergei!

**Comment by Koncopd** [comment]
@jkobject are you fine with the suggestion?

**Comment by jkobject** [comment]
Works with me!

**Comment by Koncopd** [comment]
done here https://github.com/laminlabs/lamindb/pull/1504

**Pull request closed without merging**